homeassistant/components/matrix/__init__.py

```diff
@@ -196,6 +198,11 @@ def handle_startup(_):
 
         self.hass.bus.listen_once(EVENT_HOMEASSISTANT_START, handle_startup)
 
+    def _is_url(self, url):
+        """Validate the URL can be parsed and at least has scheme + netloc."""
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
```

Comment:
This should probably verify that the scheme is `http` or `https` too..?

Reply:
Sorry I'm not sure if I follow.  This should verify that the scheme and netloc portions of the URL are valid in the format `<scheme>://<netloc>/<path>;<params>?<query>#<fragment>`.  Are you saying it should check that it's *only* an `http` or `https` scheme?  I haven't tested any other schemes myself, but I don't see the need to restrict it to only those, maybe I'm wrong though.

Reply:
Yeah, sounds like that would make sense. With the additional allowlisting from HA core and `requests` not having an adapter by default for other schemes, it shouldn't probably make much of a difference in the long run, but...
```suggestion
        return result.scheme in ('http', 'https') and result.netloc
```
probably wouldn't go amiss.

Reply:
> With the additional allowlisting from HA core and requests not having an adapter by default for other schemes,
That's true.